packages/cli/src/lib/bundler/config.ts

```diff
@@ -129,6 +144,25 @@ export async function createConfig(
     }),
   );
 
+  if (process.env.EXPERIMENTAL_FORCE_REACT_DEVELOPMENT === 'true') {
```

Comment:
There are two separate places that both need to respond to this environment variable in order for this to work, which is a bit gross. It would be nice to find a neat way of consolidating the logic - the awkward thing is that the whole `optimization` top-level config key is currently handled by `lib/bundler/optimization`, so there isn't a particularly nice way of doing so.

Reply:
Found a solution for this as part of splitting this out to https://github.com/backstage/backstage/pull/25675.